code/modules/mob/living/carbon/human/update_icons.dm

```diff
@@ -299,7 +299,7 @@ var/global/list/damage_icon_parts = list()
 
 	//END CACHED ICON GENERATION.
 	stand_icon.Blend(base_icon,ICON_OVERLAY)
-	if(src.species.bodyflags & TAIL_OVERLAPPED) // If the user's species is flagged to have a tail that needs to be overlapped by limbs...
+	if(((body_accessory && istype(body_accessory, /datum/body_accessory/tail)) || !body_accessory) && species.bodyflags & TAIL_OVERLAPPED) // If the user's species is flagged to have a tail that needs to be overlapped by limbs... (having a non-tail body accessory like the snake body will override this)

```

Comment:
Is the istype okay without a check to see if body_accessory exists beforehand? Haven't had to deal with that before and it worked just fine the way it currently is. I'll test it and change if it this other way works, thanks for pointing it out